src/engine/qcommon/msg.cpp

```diff
@@ -1137,22 +1137,48 @@ void MSG_ReadDeltaEntity( msg_t *msg, const OpaqueEntityState *from, OpaqueEntit
 }
 
 template <typename T>
-static bool IsValid(const NetcodeTable& table, int size, size_t maxsize)
+static bool IsValid(Str::StringRef name, Str::StringRef tablename, const NetcodeTable& table, int size, size_t maxsize)
 {
 	if (size % NETCODE_FIELD_SIZE != 0 || size < int(offsetof(T, END)) || size > maxsize)
+	{
+		Log::Warn("gamelogic netcode table has incorrect size");
 		return false;
+	}
+
+	// all fields should be 32 bits to avoid any compiler packing issues
+	// the "number" field is not part of the field list
+	// if this assert fails, someone added a field to the entityState_t
+	// struct without updating the message fields
+	if (table.size() + 1 != size / 4)
```

Comment:
Isn't this wrong if there is a STATS_GROUP_FIELD?

Reply:
Possible. For now I just thought "yeah playerState_t doesn't match I'll look at it later", thinking it's just one field that isn't transmitted by mistake, because of this comment:

```C++ 
// any change in playerState_t should be reflected in the table in bg_misc.cpp
// (Gordon: unless it doesn't need transmission over the network, in which case it should probably go into the new pmext struct anyway)
```

Reply:
I'm not sure how to handle `STATS_GROUP_FIELD` really. It's only supported for playerState_t and it may be nice to check it only there, but I'm not sure how to check for that. Do you think I should add a boolean parameter to the `IsValid` function?

Reply:
Well, it looks the functions for transmitting player states and entity states are just big copy/pastes of each other. Maybe they could just use one function, and then there would be no differences.

Alternatively you can do the comparsion of table size to entity state size outside the IsValid function.

Reply:
Ok. I'm not going to deduplicate the netcode functions for now. For a start, I'd be happy with it working and this warning will probably just be addressed in another PR.